src/cpp/database/database.cpp

```diff
@@ -4198,6 +4198,308 @@ void Database::load_data(
     }
 }
 
+void Database::change_entity_status_of_kind(
+        const EntityId& entity_id,
+        bool active,
+        EntityKind entity_kind)
+{
+    std::shared_lock<std::shared_timed_mutex> lock(mutex_);
+
+    switch (entity_kind)
+    {
+        case EntityKind::HOST:
+        {
+            std::shared_ptr<Host> host;
+
+            for (auto host_it : hosts_)
+            {
+                if (host_it.second->id == entity_id)
+                {
+                    host = host_it.second;
+                    break;
+                }
+            }
+
+            host->active = active;
+            break;
```

Comment:
This will be done in the new `get_entity` overload.

Reply:
Until the new method is in place, we must check here too. Imagine using this method in some other place:
`change_entity_status_of_kind(100, true, HOST)`
But there is no HOST with ID 100.
`host->active` will blow because `host` is not pointing to anything.